Mathlib/Tactic/LibrarySearch.lean

```diff
@@ -116,10 +114,12 @@ def librarySearch (goal : MVarId) (lemmas : DiscrTree Name s) (required : List E
     | some (Sum.inl suggestion) => suggestions := suggestions.push suggestion
 
   pure $ some suggestions
-    where
-  /-- Verify that the instantiated goal contains each `Expr` in `required` as a sub-expression.
-  (Make sure to not reset the state before calling.) -/
-  checkRequired : MetaM Bool := return required.all (·.occurs (← instantiateMVars (.mvar goal)))
+  --   where
+  -- /-- Verify that the instantiated goal contains each `Expr` in `required` as a sub-expression.
+  -- (Make sure to not reset the state before calling.) -/
+  -- -- TODO We need to move this check into `solveByElim`, to get proper backtracking.
```

Comment:
Is this comment obsolete? It looks like you are using `cfg.requireUsingAll` above.